### DISCUSSION TITLE
Table Headers to be in the <thead> tag

### DISCUSSION THREAD

**Discussion by parmeet1099** [ueberdosis/tiptap#3949] [General Questions]
Hi, I have been using weasyprint to convert HTML to pdf and I wanted to repeat the headers of the table when they get split onto new page in the pdf. Weasyprint respects the 'thead' tag headers and repeat them on the new page when the table get split. 
But by default tiptap renders the table rows in 'tbody' tag and even in toggleHeaderRow() it doesn't make a new <thead> tag rather it uses 'th' to make table headers.
Is there any way to achieve what i want. any guidance would help. thanks :)

**Comment by codemzy** [comment]
I use regex to do this before sending the html to the PDF creator...

```js
// add markup for thead
let pdfContent = content.replace(/<tbody><tr>(?<contents>.+?)<\/tr>/gm, function(match, contents) {
  return !contents.match(/<\/td>/gm) ? `<thead><tr>${contents}</tr></thead><tbody>` : match; // only if no td in first row
});
```

This checks if the first row only contains `th` and if it is all `th` cells then it wraps the top row in the `thead` tag and starts the `tbody` after. This is good for my use case because sometimes I have a mixture of `th` and `td` in rows so not every table needs a `thead` so it only changes those that do!
